payment/src/test/java/org/killbill/billing/payment/provider/MockPaymentProviderPlugin.java

```
@@ -339,6 +347,14 @@ public PaymentTransactionInfoPlugin refundPayment(final UUID kbAccountId, final
     }
 
     private PaymentTransactionInfoPlugin getPaymentTransactionInfoPluginResult(final UUID kbPaymentId, final UUID kbTransactionId, final TransactionType type, final BigDecimal amount, final Currency currency, final Iterable<PluginProperty> pluginProperties) throws PaymentPluginApiException {
+        if (makePluginWaitSomeMilliseconds.get() > 0) {
+            try {
+                Thread.sleep(makePluginWaitSomeMilliseconds.get());
+            } catch (InterruptedException e) {
+                e.printStackTrace();
```

You need to reset the interrupt flag. Also, I would re-raise the exception.